src/patternfly/components/Popover/popover.scss

```diff
@@ -47,30 +50,31 @@
   --#{$popover}__arrow--m-left--Rotate: 45deg;
 
   // Close
-  --#{$popover}__close--Top: calc(var(--#{$popover}__content--PaddingTop) - var(--#{$pf-global}--spacer--form-element)); // align top of button with top of text
-  --#{$popover}__close--Right: calc(var(--#{$popover}__content--PaddingRight) - var(--#{$pf-global}--spacer--md)); // align right of content
-  --#{$popover}__close--sibling--PaddingRight: var(--#{$pf-global}--spacer--2xl);
+  --#{$popover}__close--Top: calc(var(--#{$popover}__content--PaddingTop) - var(--pf-t--global--spacer--sm)); // align top of button with top of text
+  --#{$popover}__close--Right: calc(var(--#{$popover}__content--PaddingRight)); // align right of content
```

Comment:
For the Top var, there is no form element token so I'm using the sm spacer token for now. If there's a better alternative I can update.

For the Right var, I just removed the calculation as with it (after updating to the new spacer--md token) the button looked too far to the right:

![Popover close button aligned further to the right](https://github.com/patternfly/patternfly/assets/70952936/e267d9e3-e3ec-4416-9274-9c0ad402afd3)